modules/kubernetes-addons/prometheus/locals.tf

```diff
@@ -72,10 +47,50 @@ locals {
     operating_system = "linux",
   })]
 
+  amazon_prometheus_workspace_id           = var.amazon_prometheus_workspace_id == null && var.enable_amp_for_prometheus ? aws_prometheus_workspace.amp_workspace[0].id : var.amazon_prometheus_workspace_id
+  amp_workspace_url                        = var.enable_amp_for_prometheus && local.amazon_prometheus_workspace_id != null ? "https://aps-workspaces.${data.aws_region.current.id}.amazonaws.com/workspaces/${local.amazon_prometheus_workspace_id}/api/v1/remote_write" : null
+  amazon_prometheus_ingest_iam_role_arn    = var.enable_amp_for_prometheus ? module.irsa.*.ingest.irsa_iam_role_arn[0] : null
+  amazon_prometheus_ingest_service_account = local.irsa_config.ingest.service_account
+
+  amp_config_values = var.enable_amp_for_prometheus ? [{
+    name  = "serviceAccounts.server.name"
+    value = local.amazon_prometheus_ingest_service_account
+    },
+    {
+      name  = "serviceAccounts.server.create"
+      value = false
+    },
+    {
+      name  = "serviceAccounts.server.annotations.eks\\.amazonaws\\.com/role-arn"
+      value = local.amazon_prometheus_ingest_iam_role_arn
+    },
+    {
+      name  = "server.remoteWrite[0].url"
+      value = local.amp_workspace_url
+    },
+    {
+      name  = "server.remoteWrite[0].sigv4.region"
+      value = data.aws_region.current.id
+  }] : []
+
+  irsa_config = {
```

Comment:
IMHO this config is necessary.  Just makes the code more confusing to read.  Just create two Irsa module blocks.

Reply:
I thought first it may not be nice to duplicate IRSA module for each service account. I thought this is more cleaner.

Reply:
removed and duplicated the irsa modules now